Simplify band mosaic procedure

jgrss opened the pull request:
## What is this PR changing?
Following #237:
* Single-band mosaics fail because the `DataArray` reductions remove the band dimension and flip the coordinates.
* This PR replaces previous logic with simplified reductions that: 
  * keep the band dimension
  * ignore 'no data' values during the reductions

mmann1123 reviewed (approved):
Added relevant tests. Please double check my use of tempfile. I want to make sure it will work on osx and windows as well

jgrss commented on src/geowombat/backends/xarray_.py:
```diff
@@ -455,97 +455,33 @@ def mosaic(
-                                (darray + darrayb) / 2.0,
-                            ),
-                        )
+                    darray = stack.mean(dim='band', skipna=True, keepdims=True)
```
Currently, the way that mosaicking is setup is to iterate image by image and merge. So I think the median might not be too useful at the moment.

jgrss commented:
> Added relevant tests. Please double check my use of tempfile. I want to make sure it will work on osx and windows as well

Added a few tests to your additions and moved the save test.

mmann1123 reviewed (commented):
@jgrss I think the issues i was having had to do with hitting memory limits, mosaicing huge 18 band images must have been bonking it.  I've tried to recreate the issues on reasonably large datasets and haven't had the same issue.  I think this is working well for almost all purposes.

mmann1123 commented:
Oops wrong PR
